src/main/java/net/minecraftforge/common/data/AdvancementBuilderHelper.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.data;
+
+import com.google.common.collect.Maps;
+import net.minecraft.advancements.Advancement;
+import net.minecraft.advancements.AdvancementRewards;
+import net.minecraft.resources.ResourcePackType;
+
+public class AdvancementBuilderHelper {
+
+    public static boolean build (ExistingFileHelper fileHelper, Advancement.Builder builder) {
+        return builder.canBuild((advancementId) -> {
+            if (fileHelper.exists(advancementId, ResourcePackType.SERVER_DATA, ".json", "advancements")) {
+                return new Advancement(advancementId, null, null, AdvancementRewards.EMPTY, Maps.newHashMap(), null);
+            }
+            return null;
+        });
```

Comment:
When talking about the other comment. This is pretty much what I mean since you can do this and build the advancement in one method. You can ever return a more detailed error message specific to data providers.